src/main/kotlin/org/rust/ide/documentation/RustDocumentationProvider.kt

```diff
@@ -1,13 +1,46 @@
 package org.rust.ide.documentation
 
 import com.intellij.lang.documentation.AbstractDocumentationProvider
+import com.intellij.psi.PsiComment
 import com.intellij.psi.PsiElement
-import org.rust.lang.core.psi.RustFnItem
-import org.rust.lang.core.psi.RustPatBinding
+import com.intellij.psi.PsiWhiteSpace
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.util.text.MarkdownUtil
+import com.petebevin.markdown.MarkdownProcessor
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.impl.RustImplMethodMemberImpl
 import org.rust.lang.core.psi.impl.mixin.isMut
+import java.util.*
 
 class RustDocumentationProvider : AbstractDocumentationProvider() {
 
+    override fun generateDoc(element: PsiElement?, originalElement: PsiElement?): String? {
+        when (element) {
```

Comment:
This overall `when` construct ends with `return` in each possible branch, so it might be nice to rewrite this into return expression (`return when (element) {...}`), and because this is the only expression in the whole method as `fun generateDoc(...) = when (element) {...}`).